Source/DafnyCore/Triggers/TriggersCollector.cs

```diff
@@ -390,6 +390,37 @@ internal class TriggersCollector {
       return new TriggerAnnotation(children_contain_killers || expr_is_killer, new_term.Variables, collected_terms);
     }
 
+    private IEnumerable<TriggerTerm> CollectInSetOperations(Expression new_expr, Expression original_expr) {
+      yield return new TriggerTerm { Expr = new_expr, OriginalExpr = original_expr, Variables = CollectVariables(original_expr) };
```

Comment:
I don't understand the role of `OriginalExpr`, but I note that the previous L384  added just `new_expr`, not a `TriggerTerm` wrapped around `new_expr`. Is that as intended?

Reply:
> I note that the previous L384 added just new_expr, not a TriggerTerm wrapped around new_expr. Is that as intended?

I'm not sure of what you see but here is the diff I see on my side.


```diff
         if (!children_contain_killers) {
           // Add only if the children are not killers; the head has been cleaned up into non-killer form
-          collected_terms.Add(new_term);
+          collected_terms.AddRange(CollectInSetOperations(new_expr, expr));
         }
```

So it looks like we were previously adding the same thing that I add now, except that now I generate more triggers. Could you elaborate a bit more?

Reply:
My point was that, previously, `collected_terms.Add(new_term);` adds `new_term`, where the new `collected_terms.AddRange(CollectInSetOperations(new_expr, expr));` adds

* nothing, or
* `new TriggerTerm { Expr = newExpr, OriginalExpr = originalExpr, Variables = CollectVariables(newExpr) }` (which is the thing that has a `TriggerTerm` wrapped around the `newExpr`), or
* that `new TriggerTerm...` plus additional triggers.

This is probably fine. I'm just clarifying what I had tried to comment on.